lib/puppet/provider/esx_reconfigureha/default.rb

```diff
@@ -0,0 +1,44 @@
+# Copyright (C) 2013 VMware, Inc.
+provider_path = Pathname.new(__FILE__).parent.parent
+require File.join(provider_path, 'vcenter')
+
+Puppet::Type.type(:esx_reconfigureha).provide(:esx_reconfigureha, :parent => Puppet::Provider::Vcenter) do
+  @doc = "Reconigure HA Agent"
+  def create
+    begin
+      if host == nil
+        raise Puppet::Error, "An invalid host name or IP address is entered. Enter the correct host name and IP address."
+      else
+        Puppet.notice 'Reconfiguring HA Agent'
+        connection_state = host.summary.runtime.dasHostState.state if host.summary.runtime.dasHostState
```

Comment:
I must be missing something. Where is it retrieving the host from the API in order to get host.summary.runtime.dasHostState.state ?